payment/src/main/java/net/optile/payment/ui/list/NetworkCardViewHolder.java

```diff
@@ -56,57 +49,44 @@ static ViewHolder createInstance(ListAdapter adapter, NetworkCard networkCard, V
         return new NetworkCardViewHolder(adapter, view, networkCard);
     }
 
-    private void addNetworkLogos(View parent, NetworkCard networkCard) {
-        List<String> names = new ArrayList<>();
-        List<PaymentNetwork> networks = networkCard.getPaymentNetworks();
-
-        for (PaymentNetwork network : networks) {
-            names.add(network.getCode());
-        }
-        addLogoViews(parent, names);
-    }
-
     void onBind(PaymentCard paymentCard) {
-
         if (!(paymentCard instanceof NetworkCard)) {
             throw new IllegalArgumentException("Expected Networkcard in onBind");
         }
         super.onBind(paymentCard);
         NetworkCard networkCard = (NetworkCard) paymentCard;
         PaymentNetwork network = networkCard.getVisibleNetwork();
-        int size = networkCard.getPaymentNetworkSize();
-        PaymentUtils.setTestId(itemView, "card", size == 1 ? "network" : "group");
         title.setText(networkCard.getLabel());
-        bindLogos(networkCard);
+
+        if (networkCard.getPaymentNetworkSize() == 1) {
+            bindLogoView(networkCard.getCode(), networkCard.getLink("logo"));
+            bindPaymentNetwork("network", network);
+        } else {
+            logoView.setImageResource(R.drawable.ic_card);
+            bindPaymentNetwork("group", network);
+        }
+    }
+
+    private void bindPaymentNetwork(String testId, PaymentNetwork network) {
+        PaymentUtils.setTestId(itemView, "card", testId);
         bindRegistrationWidget(network);
         bindRecurrenceWidget(network);
     }
```

Comment:
This functional separation still feels a bit "uneven" - what do you think about this one:
```suggestion
        if (networkCard.getPaymentNetworkSize() == 1) {
            bindLogoView(networkCard.getCode(), networkCard.getLink("logo"));
            setTestId("network");
        } else {
            logoView.setImageResource(R.drawable.ic_card);
            setTestId("group");
        }

        bindRegistrationWidget(network);
        bindRecurrenceWidget(network);
    }

    private void setTestId(String testId) {
        PaymentUtils.setTestId(itemView, "card", testId);
    }
```

Reply:
yes, that is a lot better, will change it.